cpp/core/operators/shuffle/splitter.cc

```diff
@@ -434,20 +434,25 @@ arrow::Status Splitter::CacheRecordBatch(int32_t partition_id, const arrow::Reco
   int64_t raw_size = batch_nbytes(batch);
   raw_partition_lengths_[partition_id] += raw_size;
   auto payload = std::make_shared<arrow::ipc::IpcPayload>();
+  int64_t batch_compress_time = 0;
 #ifndef SKIPCOMPRESS
   if (batch.num_rows() <= (uint32_t)options_.batch_compress_threshold) {
     TIME_NANO_OR_RAISE(
-        total_compress_time_, arrow::ipc::GetRecordBatchPayload(batch, tiny_bach_write_options_, payload.get()));
+        batch_compress_time, arrow::ipc::GetRecordBatchPayload(batch, tiny_bach_write_options_, payload.get()));
   } else {
     TIME_NANO_OR_RAISE(
-        total_compress_time_, arrow::ipc::GetRecordBatchPayload(batch, options_.ipc_write_options, payload.get()));
+        batch_compress_time, arrow::ipc::GetRecordBatchPayload(batch, options_.ipc_write_options, payload.get()));
   }
 #else
   // for test reason
   TIME_NANO_OR_RAISE(
-      total_compress_time_, arrow::ipc::GetRecordBatchPayload(*batch, tiny_bach_write_options_, payload.get()));
+      batch_compress_time, arrow::ipc::GetRecordBatchPayload(*batch, tiny_bach_write_options_, payload.get()));
```

Comment:
I am not quite familiar with this part of code. But at first sight, it seems batch compress time should not be collected for non-compress case, i.e., `SKIPCOMPRESS` is defined or batch row num is lower than `batch_compress_threshold`. Am I missing something?
cc @zhouyuan @FelixYBW

Reply:
@PHILO-HE  This part does include compressed time and uncompressed time. The optimization method can choose to change the name directly, or add a metric to record the non-compression time.
cc @zhouyuan @FelixYBW

Reply:
> I am not quite familiar with this part of code. But at first sight, it seems batch compress time should not be collected for non-compress case, i.e., `SKIPCOMPRESS` is defined or batch row num is lower than `batch_compress_threshold`. Am I missing something? cc @zhouyuan @FelixYBW

compress time is the time to create array payload

Reply:
In the process of creating array payload, it will decide whether to compress according to the configuration of IpcWriteOptions, such as ```arrow::ipc::GetRecordBatchPayload(*batch, tiny_bach_write_options_, payload.get()));``` In this line of code,  ```tiny_bach_write_options_.codec = nullptr```; means no compression. ```arrow::ipc::GetRecordBatchPayload(batch, options_.ipc_write_options, payload.get()));```In this line of code,  ```options_.ipc_write_options.codec = CreateArrowIpcCodec(compressed_type); ```represents the compression algorithm specified by Gluten.  So I think it is not accurate to use ```total_compress_time_``` to record these two kinds of time. It is recommended to change the name, such as ```total_create_payload_time_```.
cc @FelixYBW

Reply:
Actually if there is no compression, arrow IPC just use the data buffer address to create the payload structure. There is no memcpy there. So 99% of this time is compression time. "totaltime compress" is OK.

@zhztheplayer did we add decompress time in reducer side? currently decompress is running in task thread, right?